asyncapi/document.go

```diff
@@ -2,17 +2,150 @@ package asyncapi
 
 // Document is an object representing an AsyncAPI document.
 // It's API implements https://github.com/asyncapi/parser-api/blob/master/docs/v1.md.
+// NOTE: this interface is not completed yet.
 type Document interface {
 	Extendable
+	Channels() []Channel
+	HasChannels() bool
+	ApplicationPublishableChannels() []Channel
+	ApplicationPublishableMessages() []Message
+	ApplicationPublishOperations() []Operation
+	ApplicationSubscribableChannels() []Channel
+	ApplicationSubscribableMessages() []Message
+	ApplicationSubscribeOperations() []Operation
+	ClientPublishableChannels() []Channel
+	ClientPublishableMessages() []Message
+	ClientPublishOperations() []Operation
+	ClientSubscribableChannels() []Channel
+	ClientSubscribableMessages() []Message
+	ClientSubscribeOperations() []Operation
+	Messages() []Message
 	Server(name string) (Server, bool)
 	Servers() []Server
 	HasServers() bool
 }
 
+// Channel is an addressable component, made available by the server, for the organization of messages.
+// Producer applications send messages to channels and consumer applications consume messages from channels.
+type Channel interface {
+	Extendable
+	Identifiable
+	Path() string // Path is the identifier
+	Description() string
+	HasDescription() bool
```

Comment:
You should create something like `Descriptionable` (or better name 😄 ) struct for `Description` and `HasDescription` functions and use it as `Extendable` struct (here and in other places) - something like [mixins](https://github.com/asyncapi/parser-js/blob/master/lib/mixins/description.js) in JS parser

Reply:
Done!